### PR TITLE
DMP-4959: Previously is_current=true audio is not set to false when new version added

### PR DESCRIPTION
**Pull request by Ben-Edwards-cgi**
### Change description ###
# Summary of Git Diff

The diff reflects updates made to the integration test for the `AudioControllerAddAudioMetadata` and related services in the DARTS audio module of a Java application. Key enhancements include the addition of a new test to verify the behavior of audio metadata addition when dealing with duplicate media entries, as well as minor adjustments and imports in the associated classes.

## Highlights

- **New Test Case Added**: 
  - A new test method `whenAddAudioIsCalled_withTheLastCurrentValueNotBeingTheLastAntecendntValue_shouldUseLastCurrentValueForDuplicateChecks` was added to validate duplicate handling in audio metadata addition.
  
- **Imports Updated**:
  - New imports for `AudioUploadService`, `TriConsumer`, and `Function` were added to the test class.
  - The `assertThat` assertion method from AssertJ was introduced for better readability in assertions.

- **Service Layer Changes**:
  - In the `AudioUploadServiceImpl`, changes were made to how duplicate media files are handled, specifically filtering for current media entities when checking for duplicates.
  - An additional check was added to return only media entries marked as current if available.

- **Repository Layer Changes**:
  - Minor formatting adjustments were made to the `MediaRepository` interface for consistency.

This diff enhances the robustness of audio metadata handling in the DARTS application by ensuring that the system properly manages duplicate audio entries, improving overall stability and reliability.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/darts/audio/service/impl/AudioUploadServiceImpl.java [hunk 1]
```diff
@@ -204,7 +203,15 @@ private List<MediaEntity> getLatestDuplicateMediaFiles(AddAudioMetadataRequest a
             addAudioMetadataRequest.getStartedAt(),
             addAudioMetadataRequest.getEndedAt());
 
-        // now lets get the lowest level media objects so that they can act as a basis for the antecedent
```
Comment (jackmaloney) [inline src/main/java/uk/gov/hmcts/darts/audio/service/impl/AudioUploadServiceImpl.java:207]: Is the comment not still relevant?

File: src/integrationTest/java/uk/gov/hmcts/darts/audio/controller/AudioControllerAddAudioMetadataIntTest.java [hunk 2]
```diff
@@ -496,6 +503,91 @@ void shouldHideIncomingMediaAndCopyExistingAdminAction_whenIncomingMediaHasExist
...
+            media.setIsCurrent(false);
+            return dartsDatabase.save(media);
+        };
+        TriConsumer<Integer, MediaEntity, MediaEntity> alignDataMediaEntity = (chronicleId, previousMedia, newMediaEntity) -> {
+            newMediaEntity.setChronicleId(String.valueOf(chronicleId));
+            newMediaEntity.setAntecedentId(String.valueOf(previousMedia.getId()));
+        };
+
+        /*
+        Values pulled from staging which caused the origional failure (These values are replicated in this test)
```
Comment (davet1985) [inline src/integrationTest/java/uk/gov/hmcts/darts/audio/controller/AudioControllerAddAudioMetadataIntTest.java:534]: s/origional/original

File: src/main/java/uk/gov/hmcts/darts/audio/service/impl/AudioUploadServiceImpl.java [hunk 3]
```diff
@@ -204,7 +203,15 @@ private List<MediaEntity> getLatestDuplicateMediaFiles(AddAudioMetadataRequest a
...
+        List<MediaEntity> currentDuplicates = identicalMediaEntities
+            .stream()
+            .filter(media -> Boolean.TRUE.equals(media.getIsCurrent()))
+            .toList();
+
+        //Return only is_current true items if there are any
+        if (!currentDuplicates.isEmpty()) {
+            return currentDuplicates;
+        }
         Tree<MediaEntityTreeNodeImpl> tree = new Tree<>();
```
Comment (jackmaloney) [inline src/main/java/uk/gov/hmcts/darts/audio/service/impl/AudioUploadServiceImpl.java:215]: Is this block (L215 to 219) still relevant? In what situation what it be hitting this?

File: src/main/java/uk/gov/hmcts/darts/audio/service/impl/AudioUploadServiceImpl.java (hunk 1 above)
Comment (Ben-Edwards-cgi) [reply src/main/java/uk/gov/hmcts/darts/audio/service/impl/AudioUploadServiceImpl.java:207]: I moved have re-added this comment and moved it to the correct place

File: src/main/java/uk/gov/hmcts/darts/audio/service/impl/AudioUploadServiceImpl.java (hunk 3 above)
Comment (Ben-Edwards-cgi) [reply src/main/java/uk/gov/hmcts/darts/audio/service/impl/AudioUploadServiceImpl.java:215]: This is a catch all, so if for what ever reason a duplicate entry is found but it is not the current value it can still get picked up

File: src/integrationTest/java/uk/gov/hmcts/darts/audio/controller/AudioControllerAddAudioMetadataIntTest.java (hunk 2 above)
Comment (Ben-Edwards-cgi) [reply src/integrationTest/java/uk/gov/hmcts/darts/audio/controller/AudioControllerAddAudioMetadataIntTest.java:534]: Updated

**Comment by davet1985** [comment]
I've tested this on the dev environment and although it's fixed the issue with `is_current`, the antecedent_id of the newly added audio version, with med_id `165106`, is set to `41037` which is not correct. It should be `163965` as this was the previous version of the audio that was received.

It doesn't matter which one is current when setting the antecedent_id, that should always be the last one received, ordering by created_ts descending.

Screenshot 2025-04-09 at 11 24 00

**Review by davet1985 — APPROVED** [review]
LGTM

**Review by jackmaloney — APPROVED** [review]
